visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/DomainClientLayer/Auth/Rethrower.java

```diff
@@ -0,0 +1,30 @@
+package com.petclinic.visits.visitsservicenew.DomainClientLayer.Auth;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+import java.util.function.Function;
+
+@RequiredArgsConstructor
+@Component
+public class Rethrower {
+    private final ObjectMapper objectMapper;
+    public Mono<? extends Throwable> rethrow(ClientResponse clientResponse, Function<Map, ? extends Throwable> exceptionProvider) {
+        return clientResponse.createException().flatMap(n ->
+        {
+            try {
+                final Map map =
+                        objectMapper.readValue(n.getResponseBodyAsString(), Map.class);
+                return Mono.error(exceptionProvider.apply(map));
+            } catch (JsonProcessingException e) {
+//                e.printStackTrace();
+                return Mono.error(e);
+            }
+        });
+    }
```

Comment:
Can you clarify what this method is doing and why you need it?

Reply:
It rethrows error to what called it. I need it because of how the authServiceClient was implemented(copied from the api-gateway).